### PR TITLE
Fix the CompilerPluginManager for LS

### LINKED ISSUE #43060: [Bug]: Compiler plugins are not loaded properly  when there is a change in import for LS side.
### Description

When there is an import change a new package instance is created. But, the project instance is kept the same.  CompilerpluginContext stored in the project(https://github.com/ballerina-platform/ballerina-lang/blob/b8fd113bfdb7c5dfa71fa22f4559399739f880a0/compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java#L54) is also kept without clearing. This is returned when a new compiler plugin instance is created at (https://github.com/ballerina-platform/ballerina-lang/blob/b8fd113bfdb7c5dfa71fa22f4559399739f880a0/compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java#L195). So, the compiler plugin from the new import won't be loaded. 

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
Fixes #43060 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java [hunk 1]
```diff
@@ -51,6 +51,12 @@ private CompilerPluginManager(PackageCompilation compilation,
     }
 
     static CompilerPluginManager from(PackageCompilation compilation) {
+        // Skip initialization if the compiler plugins are already initialized for the project
+        if (!compilation.packageContext().project().compilerPluginContexts().isEmpty()) {
+            List<CompilerPluginContextIml> compilerPluginContexts =
+                    compilation.packageContext().project().compilerPluginContexts();
```
Comment (Thevakumar-Luheerathan) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java:57]: @azinneera  Shall we try to move this `compilerPluginContexts` from `Project` to `PackageContext` something like `PackageResolution`. I feel that's more relevant. WDYT?

Comment (gayaldassanayake) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java:57]: +1

Comment (Dilhasha) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java:57]: Shall we create a separate issue for this?

Comment (Thevakumar-Luheerathan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/CompilerPluginManager.java:57]: sure will do..

**Review by Dilhasha — APPROVED** [review]

**Pull request merged**